VideoPress:  add RN mobile video player block

jhnstn opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/5509

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* Add a native supported VideoPress video player block

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->
TBD

jhnstn committed:
[not verified] Add native supported player component

jhnstn committed:
[not verified] Render native video player block

jhnstn committed:
[not verified] Remove unused native style sheet

retrofox committed:
VideoPress: add Poster panel to video block sidebar (#29150)

* [not verified] first approach of the poster panel

* [not verified] export poster media types constant

* [not verified] tweak poster image help section

* [not verified] set background image to panel button

* [not verified] add poster panel to the default settings tab

* [not verified] changelog

* Update projects/packages/videopress/src/client/block-editor/blocks/video/components/poster-panel/style.scss

Co-authored-by: Douglas Henri <dhasilva@users.noreply.github.com>

* [not verified] move poster action label to a const

* two const > one const :smart:

---------

Co-authored-by: Douglas Henri <dhasilva@users.noreply.github.com>
Co-authored-by: Douglas <douglas.henri@automattic.com>

retrofox committed:
VideoPress: tweak Remove poster button (#29157)

* [not verified] use tertiary and destructive style for remove btn

* [not verified] do not use CSS module

* [not verified] tweak delete button layout

* changelog

retrofox committed:
VideoPress: set poster height in block sidebar based on video ratio (#29173)

* [not verified] use a const to define image styles dynamically

* [not verified] adjust poster image height based on video ratio

* changelog

thingalon committed:
[Boost] Exclude Defer JS in the customizer preview (#29143)

* Exclude deferred JS in customizer preview

* changelog

---------

Co-authored-by: Mark George <thingalon@gmail.com>

vikneshwar committed:
Revise Jetpack connection agreement text to comply with our User Agreement (#28403)

Co-authored-by: Matt Gawarecki <matt.gawarecki@automattic.com>

fluiddot commented on projects/packages/videopress/src/client/block-editor/blocks/video/edit.native.tsx:
```diff
@@ -39,6 +56,28 @@ export default function VideoPressEdit( { attributes, setAttributes } ): React.R
+	// Pick video properties from preview.
+	const { html: previewHtml, scripts } = preview;
+
+	const html = previewHtml || cacheHtml;
```
Regarding `cacheHtml` attribute, we'd need to replicate the same logic we have in the web version ([reference](https://github.com/Automattic/jetpack/blob/36fde37a4444d1d1debcef9d5b3bae8951aa1dc7/projects/packages/videopress/src/client/block-editor/blocks/video/edit.tsx#L160-L177)).

jhnstn replied:
I'm not sure this is entirely true. 

I did see the effect that sets the cacheHtml attribute in the web version, which you referenced,  but I don't see where the attribute is actually being used in the video player component.  For now I would suggest we keep the current logic scoped to what is needed to render the video player.

That said I find the comments above the effect hook puzzling and makes wonder if the `cacheHtml` attribute is actually needed. The comment mentions setting [`html` and `thumbnail` attributes](https://github.com/Automattic/jetpack/blob/36fde37a4444d1d1debcef9d5b3bae8951aa1dc7/projects/packages/videopress/src/client/block-editor/blocks/video/edit.tsx#L162), neither of which are actually set in the effect hook. 
I'll do some commit spelunking to find the reasoning for the setting the `cacheHtml` attribute, perhaps it is needed for the some of the other components like `<VideoPressUploader>` or `<PlaceholderWrapper>`

fluiddot replied:
> I did see the effect that sets the cacheHtml attribute in the web version, which you referenced, but I don't see where the attribute is actually being used in the video player component. For now I would suggest we keep the current logic scoped to what is needed to render the video player.

I think the purpose of the `cacheHtml` attribute is to be used upon the component being mounted to display the video player while the preview is being fetched, as it's mentioned in this comment:
https://github.com/Automattic/jetpack/blob/46d3698a631903c49531e264dcb18f35816d7d40/projects/packages/videopress/src/client/block-editor/blocks/video/edit.tsx#L184-L186

I understand that this is an enhancement to avoid displaying a loading indicator, which we might consider implementing in the native version. In any case, I tested this functionality in the web version and it's true that is hard to tell the difference with and without having the cache, especially as the HTML content is an iframe that still needs to load the video player content 🤔.

jhnstn replied:
> I think the purpose of the cacheHtml attribute is to be used upon the component being mounted to display the video player while the preview is being fetched

Seems like the `const html  = previewHtml || cacheHtml` assignment covers the mounting period in the block lifecycle. 

This enhancement might be for when the preview is fetched _after_ the component is mounted. The hook logic suggests that the `previewHtml` would have to exist then not exist before the `setAttribute` call would have any relevance.  

I'm guessing that happens when the video source is updated. If so, we should probably handle that when we get to settings or uploading updates.

fluiddot commented on projects/packages/videopress/src/client/block-editor/blocks/video/components/videopress-player/index.native.js:
```diff
@@ -0,0 +1,57 @@
+	return (
+		<View
+			style={ { height: videoPlayerHeight } }
+			onLayout={ ( { nativeEvent } ) => scaleVideoPlayer( nativeEvent ) }
```
I recall that the height was automatically set in the `Sandbox` component ([reference](https://github.com/WordPress/gutenberg/blob/b4e9e0aed4910eb7b6f57b67f866b37ea05e78ee/packages/components/src/sandbox/index.native.js#L264-L281)) communicating with the embed content, not sure if we could apply a similar approach for the VideoPress player 🤔 .

jhnstn replied:
I _believe_ I was running into issues with the aspect ratio of the videos, not just the height. I'll play around with this to see if I can get the Sandbox to handle it correctly

jhnstn replied:
Ok I figured out what's going on here. I was following the web version a bit too closely I think. There is a "loading" view that needs some height.  I think the web version also needs to handle responsive UI changes so I'm assuming the height needs to be handled explicitly.

I can handle the "is loading" state better which simplifies a lot of things. 
Thanks for pointing that out!

fluiddot commented on projects/packages/videopress/src/client/block-editor/blocks/video/components/videopress-player/index.native.js:
```diff
@@ -0,0 +1,57 @@
+	preview,
+	isRequestingEmbedPreview,
+} ) {
+	const [ videoPlayerHeight, setVideoPlayerHeight ] = useState( 250 );
```
[In the web version, a temporary height is set](https://github.com/Automattic/jetpack/blob/36fde37a4444d1d1debcef9d5b3bae8951aa1dc7/projects/packages/videopress/src/client/block-editor/blocks/video/components/videopress-player/index.js#L66-L73), maybe we could replicate a similar logic instead of having a constant, WDYT?

jhnstn replied:
Are you referencing the magic number `250` ?  That's just a replacement for the temporary styling height that was added [here](https://github.com/Automattic/jetpack/pull/29165/files#diff-a48bfa252db3a7399b5a161b6b36a5b88c3db3d37780daece46693f418238647L12). I would expect that to be modified later once we get into refining the component.  That said the web version [also uses a constant](https://github.com/Automattic/jetpack/blob/36fde37a4444d1d1debcef9d5b3bae8951aa1dc7/projects/packages/videopress/src/client/block-editor/blocks/video/components/videopress-player/index.js#L189), just not in the `useState` call. ( Just noticed while writing this that the web version now uses [a constant as well](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/videopress/src/client/block-editor/blocks/video/components/player/index.tsx#L71) )

Or are you talking the web version use of  `state.videoPlayerTemporaryHeight` . It's really just naming.  I've simplified the logic involved in calling `setVideoPlayerHeight` vs the web version `setVideoPlayerTemporaryHeight` but the intent is the same. I expect the calling logic to become more complex in future changes so I opted to keep it simple for now. 
I found the use of _temporary_ misleading in the web version. I couldn't find a condition where the "temporary" height is not used, it is ignored once the embed is rendered but it's not replaced or removed. Perhaps a better name, in both cases, is `playerWrapperHeight` since that is where how the state property is [used](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/videopress/src/client/block-editor/blocks/video/components/player/index.tsx#L201).

fluiddot replied:
> Are you referencing the magic number `250` ? That's just a replacement for the temporary styling height that was added [here](https://github.com/Automattic/jetpack/pull/29165/files#diff-a48bfa252db3a7399b5a161b6b36a5b88c3db3d37780daece46693f418238647L12). I would expect that to be modified later once we get into refining the component. That said the web version [also uses a constant](https://github.com/Automattic/jetpack/blob/36fde37a4444d1d1debcef9d5b3bae8951aa1dc7/projects/packages/videopress/src/client/block-editor/blocks/video/components/videopress-player/index.js#L189), just not in the `useState` call. ( Just noticed while writing this that the web version now uses [a constant as well](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/videopress/src/client/block-editor/blocks/video/components/player/index.tsx#L71) )

Yeah, I was referencing the magic number. Oh, true, in the web version we also have a constant. In that case, omit my comment then, thanks for elaborating!

> Or are you talking the web version use of `state.videoPlayerTemporaryHeight` . It's really just naming. I've simplified the logic involved in calling `setVideoPlayerHeight` vs the web version `setVideoPlayerTemporaryHeight` but the intent is the same. I expect the calling logic to become more complex in future changes so I opted to keep it simple for now.
> I found the use of _temporary_ misleading in the web version. I couldn't find a condition where the "temporary" height is not used, it is ignored once the embed is rendered but it's not replaced or removed. Perhaps a better name, in both cases, is `playerWrapperHeight` since that is where how the state property is [used](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/videopress/src/client/block-editor/blocks/video/components/player/index.tsx#L201).

Based on the following comment, I understand that the temporary height is just an initial calculation we make based on the block's width and the video's aspect ratio. Once the player is loaded, the height is replaced with the final value, so it seems the temporariness is based on the fact that is only used during the loading state.

https://github.com/Automattic/jetpack/blob/8f7114bde33077e8b45940ac579126c4d77a2ac8/projects/packages/videopress/src/client/block-editor/blocks/video/components/player/index.tsx#L60-L66

jhnstn replied:
It's true that there is an initial value for the height that is derived from the aspect ratio. 
The issue I see is the naming. From what I observed the `videoPlayerTemporaryHeight` state variable is always used to set the style height property. 

In fact once the video is available, `videoPlayerTemporaryHeight` is set to `auto` [here](https://github.com/Automattic/jetpack/blob/36fde37a4444d1d1debcef9d5b3bae8951aa1dc7/projects/packages/videopress/src/client/block-editor/blocks/video/components/videopress-player/index.js#L103) 

In other words, this state variable has an initial value to set a placeholder height, and then is set to `auto` once the actual height is available. To me that variable is not temporary. If I see something named temporary with a comment that it's eventually ignored,  I would expect to see it used like 

```
const value = actualValue || temporaryValue
```

instead the code here is more like 
```
let temporaryValue = 1;
eventually( (actualValue) => setTemporaryValue(actualValue) )
const value = temporaryValue
```

fluiddot replied:
> In fact once the video is available, `videoPlayerTemporaryHeight` is set to `auto` [here](https://github.com/Automattic/jetpack/blob/36fde37a4444d1d1debcef9d5b3bae8951aa1dc7/projects/packages/videopress/src/client/block-editor/blocks/video/components/videopress-player/index.js#L103)
> 
> In other words, this state variable has an initial value to set a placeholder height, and then is set to `auto` once the actual height is available. To me that variable is not temporary. If I see something named temporary with a comment that it's eventually ignored, I would expect to see it used like

Yep, good point. I agree the `videoPlayerTemporaryHeight ` is a bit confusing, especially when the `auto` value is set because is not consistent with the temporariness led by the variable name. I understand that the purpose of `auto` is basically disabling the temporary height and letting the video player control the height, maybe the value should be renamed to `disabled` or something similar 🤔. For the native version, if we have to provide a customized version of the component, we can discuss further how to name and structure this part if needed.

fluiddot commented on projects/packages/videopress/src/client/block-editor/blocks/video/components/videopress-player/index.native.js:
```diff
@@ -0,0 +1,57 @@
+			style={ { height: videoPlayerHeight } }
+			onLayout={ ( { nativeEvent } ) => scaleVideoPlayer( nativeEvent ) }
+		>
+			{ embedPreviewRequested && <SandBox html={ html } scripts={ scripts } /> }
```
It's likely that we'd need to pass a custom JS script to communicate with the player, especially in relation to fetching the token for private videos. As an example, [here](https://github.com/Automattic/wp-calypso/blob/9b66da6ae2297cd78b19cad5982e66a516b2ca94/client/post-editor/media-modal/detail/detail-preview-videopress.jsx#L113-L146) is how Calypso handles the token request when asked from the player.

jhnstn replied:
Agreed, however that feels like a conversation for future changes.
The scope here is to get a native version of the player working for public videos.

fluiddot replied:
Good point. I agree that if we limit the scope to public videos this won't be necessary 👍. Let's keep this in mind in the next iteration but not as a blocker of the PR.

anomiex committed:
cli: `generate` command should ask for plugin versioning method (#29078)

* cli: `generate` command should ask for plugin versioning method

We've been defaulting to semver like we do for all other project types,
but it seems this may be confusing teams working on the plugins who may
be expecting WordPress-style versioning. Instead let's ask when the
plugin is being created.

samiff committed:
jetpack-mu-wpcom: fix undefined is_plugin_active fatal on wpcom (#29158)

jeyip committed:
Launchpad: Refactor launchpad save modal (#29134)

ice9js committed:
Add page navigation for the new feedback dashboard (#28826)

Co-authored-by: Jeremy Herve <jeremy@jeremy.hu>

IanRamosC committed:
Fix/discount for first month trials (#29139)

* Add a pricing-utils helper folder

This folder will store helper functions related to pricing. For now, I've only included `isFirstMonthTrial` as it's the only one required for this PR, but intention is to add more functions related to pricing, instead of having them only as reducers/selectors.

* Recommendations: remove discount if there's a trial

* Recommendations: remove the coupon code from the URL if there's a trial

This change is made in the component instead of in withUpgradeUrl because this rule is more strict and related specifically to what is displayed in this card. This change is mostly to avoid visual confusion.

* Add changelogger

* Remove unnecessary CSS

* Fix changelogger validity

ouikhuan committed:
Replace links with Jetpack redirect service for migration plugin (#29129)

jcheringer committed:
Forms: Increase form fields padding based on user-defined border-radius (#28820)

* Increase form fields padding based on user-defined border-radius

* changelog

* Fix field padding in the Forms package

* changelog

* Fix changelogger

a8ck3n committed:
Odyssey: Remove Calypso recommendations from legacy stats page (#29115)

* Odyssey: Remove Calypso nudge.

Doesn’t make sense to push to Calypso now that Odyssey is an option.

* Odyssey: Remove Calypso nudge (2).

This time from the loading UI.

* Changelog++

oskosk committed:
Related Posts: Add block support in server block renderer (#29135)

* [not verified] Add block support for font family in Related Posts block server block renderer

* [not verified] changelog

retrofox committed:
VideoPress: remove uploading image for video poster when uploading video (#29183)

* remove image for poster when uploading video

* fix default aspect ratio

* changelog

jeherve committed:
Widget Visibility: switch to shared Analytics implementation. (#29181)

samiff committed:
Jetpack: update to-test.md for v11.9 (#29186)

* Jetpack: update to-test.md for v11.9

* Typos and formatting

anomiex committed:
build: Cleanup workflows (#29188)

nateweller committed:
WAF: Add support for handling IP ranges in allow/block lists (#29131)

ice9js committed:
Make Jetpack Forms dashboard use the "Feedback" menu item in WP Admin (#29198)

* Update the forms dashboard to use the same feedback menu item

* changelog

gmjuhasz committed:
Remove `Default Image` option from SIG panel (#29206)

* Remove default image type as it's not used yet

* changelog

jeherve committed:
Custom CSS: use slug to link to Custom CSS settings (#29202)

jeherve committed:
Blocks: update scaffolding CLI commands (#29201)

danielbachhuber committed:
Staging Sites: Add staging site attributes to the site object (#29192)

* REST API: Add staging site attributes to the site object

* Make the key name more specific

CGastrell committed:
Remove body font normalization on contact-form module (#29166)

* no need to normalize entire body and labels fonts

* changelog

* reword changelog entry

thingalon committed:
Prepare Super Cache for the next dev cycle (#29196)

* Update changelogs and readmes for 1.9.4-beta release

* Initialize next dev cycle for wp super cache

---------

Co-authored-by: Mark George <thingalon@gmail.com>
Co-authored-by: Samiff <samiff@users.noreply.github.com>

coder-karen committed:
CRM: Alphabetizing extensions and highlighting featured Woo extensions on extensions page. (#29199)

paulopmt1 committed:
Allowing getting posts by ids in GET /posts response using include array (#29149)

* Allowing getting posts by ids at GET /posts response

* changelog

* Added casting to include variable and improved its compatibilty

* Added the new filter for both v1 and v1.2 versions of the API

samiff committed:
Jetpack: 11.9-beta backport changelogs (#29211)

* Jetpack: 11.9-beta changelog

* jetpack-mu-wpcom: v1.0.1 changelog

* Jetpack: squash readme

* Backup: 1.5-beta changelog

* Social: 1.8.0-beta changelog

* Init next plugin cycles

retrofox committed:
Jetpack: register VideoPress video block based on the filtered extensions (#29207)

* [not verified] add get_extensions() public method

* [not verified] change how to register videopress video block
it gets the (filtered) extensions to check whether the  videopress/video extension is available. If so, it registers the block.

* [not verified] changelog

* [not verified] Update projects/plugins/jetpack/class.jetpack-gutenberg.php

Co-authored-by: Brad Jorsch <anomiex@users.noreply.github.com>

---------

Co-authored-by: Brad Jorsch <anomiex@users.noreply.github.com>

samiff committed:
Jetpack: 11.9-beta changelog editorial (#29212)

* Jetpack: 11.9-beta editorial

* changelog

dhasilva committed:
VideoPress: Add delete action to video details page (#29161)

* renaming prop

* add empty actions popover on details page

* changelog

* refactor dropdown adding buttons and 'add to new post' feature

* extract DeleteVideoConfirmationModal component

* add video delete confirmation modal to the video details dropdown

* add busy state to video details page

* fix warning on PopoverWithAnchor component

* bump version

lhkowalski committed:
VideoPress: support "private" as video "site default" privacy on public Atomic sites (#29104)

* Create function to determine if VideoPress videos are private by default

This is the start of the refactoring to improve how we handle privacy on Atomic sites.

* Add check for Atomic sites

If it's a private atomic site, the site default privacy setting is private as well. If it's public, treats like a Jetpack site and look for the `videopress_private_enabled_for_site` option.

* Include check for Simple sites

For Simple sites, the site privacy controls the site default video privacy. If the site is private, the default privacy for videos is private as well. The other way around is also true.

* Use the same logic from the Ajax class to determine video privacy default

* Rely on the Data class to determine site default video privacy

* Bump versions

* Fix the return value for private atomic sites

If the site is Atomic and is private, the function should return true signaling that the videos are private for the site.

* Force return value to bool type

ouikhuan committed:
Initial release for Move to WordPress.com (#29220)

- Update changelog

xristos3490 committed:
Fix out-of-bounds submenus on the admin menu (#28355)

Co-authored-by: Jeremy Herve <jeremy@jeremy.hu>

gogdzl committed:
CRM: Fix potential XSS attack in tax settings page (#29215)

* Escape variables

* changelog

lhkowalski committed:
VideoPress: Disable privacy toggle for private atomic sites (#29169)

* [not verified] Expose siteType and siteIsPrivate flags on VideoPress initialState var

* [not verified] Add changelog file

* [not verified] Disable privacy checkbox when the site is an Atomic private site

* Show message explaining why the checkbox setting is disabled

* Use constants to refer to site type on settings component

* Organize the site type into it's own type

* Use constants to refer to site type on settings hook

* Move site type to central point on the project

* Fix small visual issues

* Add i18n call on the private atomic reason message

Co-authored-by: Damián Suárez <rdsuarez@gmail.com>

* Fix broken icon position on different viewport

* Remove unnecessary call to classname function

* Set max-width for tooltip box to improve readability

---------

Co-authored-by: Damián Suárez <rdsuarez@gmail.com>

creativecoder committed:
Adds wpcom/v3/blogging-prompts REST API endpoint (#29182)

Co-authored-by: Jeremy Herve <jeremy@jeremy.hu>

creativecoder committed:
Removes unused conversation and dialogue blocks (#29210)

Co-authored-by: Jeremy Herve <jeremy@jeremy.hu>

anomiex committed:
svn auto-publish: Correctly publish first version (#29229)

Usually we leave the old version marked as stable, so humans can
smoke-test before manually updating the tag. But in the case of the
first commit there's no "old version", which causes our code that tries
to keep the old version to break.

Since there seems to be no documented way say "no version is stable
yet", it seems our best bet is to just let the version being
auto-published be marked as stable for the initial commit.

mattwiebe committed:
Launchpad: add REST API endpoint (#29082)

This contains:

1. Infra for loading the same endpoint on both Simple and Atomic.
2. A Launchpad endpoint - see #73830

For 1), we still need to map the passthrough in `jetpack-bridge.php` but perhaps that could be done automatically, for endpoints here, in the future.

davemart-in committed:
Removing awkward spacer after every 2nd button (#29228)

davemart-in committed:
Fix broken Tumblr button inside "More" button overlay (#29231)

davemart-in committed:
Added space bar press to improve "More" button a11y (#29232)

* Added space bar as an option to open More button overlay

* Removed unused 'event' prop

dhasilva committed:
VideoPress: Add chapters generation from description to dashboard (#29155)

* [not verified] increase description textarea height to 4 rows

* [not verified] add live chapters parsing to the dashboard

* [not verified] organize imports

* [not verified] add chapter generation and update on save

* [not verified] changelog

* [not verified] increase description textarea rows

Co-authored-by: Damián Suárez <rdsuarez@gmail.com>

* [not verified] extract chapter helper components

* extract fetchVideoItem function

* add handleChaptersUpdate call to handleSaveChanges on video details page

---------

Co-authored-by: Damián Suárez <rdsuarez@gmail.com>

matticbot committed:
Update wordpress monorepo (#29216)

* Update wordpress monorepo
* Remove use of unstableOnFocus for richtext component

---------

Co-authored-by: Renovate Bot <bot@renovateapp.com>
Co-authored-by: Brad Jorsch <brad.jorsch@automattic.com>

a8ck3n committed:
Stats: Update layouts on legacy page. (#29221)

* Stats: Fix for layouts on legacy page.

The updated margin values remove the large gap above the spinner and leave enough room for the entire status message to be shown when JS is disabled.

* Changelog++

retrofox committed:
VideoPress: rewrite player by using TypeScript (#29226)

* rename player filename

* rename player folder

* remove unused debug() instance

* set Player component types

* tidy component jsdoc block

* fix TS issues

* changelog

* prefix types with type

coder-karen committed:
CRM: Making sure id is included in a tag inclusion (#29209)

coder-karen committed:
CRM: Fix tax rate deletion for single entries on tax settings page. (#29227)

thingalon committed:
[Super Cache] Add mike as a contributor (#29241)

Co-authored-by: Mark George <thingalon@gmail.com>

jeherve committed:
Blocks: add a new Cookie Consent Block (#29197)

Co-authored-by: Omar Alshaker <omar@omaralshaker.com>

alshakero committed:
Tiny PR: Fix cookie consent default content and CSS (#29251)

jeherve committed:
Forms: fix permission check in Responses endpoint (#29223)

pyronaur committed:
Jetpack Boost: WIP DataSync Critical CSS (#29122)

* Add dependencies

* Improve equals method

* Replace all dashes

* Fix newly included packages

* Customize svelte store updating

* Ensure the prevValue doesn't get modified by reference

* Use AbortController to prevent rapid store update syncs

* Add changelog

* Implement WP JS DataSync to mirror Critical CSS State

* DataSync based Critical CSS Generation

* Move recommendation dismissal to DataSync

* Remove the Generator dependency

* Clean up reset state

* Cleanup & Remove unused code

* Refactor the cloud codebase

* Refactor suggestRegenerate functionality

* criticalCSS global type is now unused

* Restore polling, but via DataSync

* Clean up ProviderSources type

* remove import which is never used

* changelog

* Remove eslint exception

* Clean up compile errors by accounting for the difference between my JSONObject and the zod one

* Improve data sync entry composability

* Add Boolean_Entry helper

* Delete the old ds entry handler

* Log the failed provider key

* Use the new data-sync implementation

* Add Cloud_CSS_Sync handler

* Ensure that the handler classes are not extended

* Set pending Cloud CSS pending providers

* Allow sending "false" too

* Refactor progress tracking with Cloud CSS

* Set default value for critical_css_suggest_regenerate

* Mark generation as complete when all providers are accounted for

* Separate generation and provider states

* Get rid of event-based critical css regeneration

* Remove state from local CCSS generation fn

* requestCloudCss is now just regenerateCriticalCss anyway

* phpcs

* Merge issues into Provider state

* Update maybeGenerateCriticalCss

* Refactor error reporting

* Separate errors from fatal errors

* criticalCssState -> criticalCssStatus

* Critical CSS Status -> State

* recommendations -> errors

* -ds -> -datasync

* Move progress

* Make CritcalCssState Read-only

* More reaonable Critical CSS State boundaries

* Apparently, booleans have to be coerced to pass validation

* stopTheShow -> criticalCssFatalError

* No longer necessary to disable regeneration suggestions....

....from PHP - this is handled by TypeScript now

* Clear Critical CSS State in Invalidator

* Restore auto-generate behavior

* Dirty hacks to get past compile errors

* Fix PHP Fatal error

* Add a refactoring comment

* Merge Cloud & Local Critical CSS Handling

This needed to be done to make it easier to identify what the common parts are. The next step on this thread is to separate the concerns in to smaller reusable classes.

* Disable debugging code for a bit

* Remove a circular dependency; svelte-data-sync-client was listing itself as a dependency

* Fix linting issues

* Fix more linting errors

* fix php ignores

* changelog

* Remove unused import

* Clean up zod catch

* Update the whole local state from the server

* Set error status if generating fails

* Fix progress bar jank

* Fix progress bar jank 🌈

* Use the latest version of critical-css-gen; avoid unnecessary iframe loads

* Fix Critical CSS resuming issue

* Remove refactoring comments

---------

Co-authored-by: Mark George <thingalon@gmail.com>

sergeymitr committed:
Connection: Improve JS code in the connection owner removal notice (#29087)

* Replace jQuery code with VanillaJS
* Properly encode data in the JS
* Improve error handling

creativecoder committed:
Writing prompts: adds basic UI to blogging prompts block (#29189)

CGastrell committed:
Add response inbox app and endpoint (#29007)

* [not verified] add state and provider wrapping

* [not verified] use state data/selectors on inbox component

* [not verified] add subtitle on layout

* [not verified] trigger selection event on last checked item for now. Need new 'activeRow' thing and its related events/classes

* [not verified] add changelog entries

* [not verified] Add jest to @automattic/jetpack-forms to enable unit testing JS

* [not verified] Add @wordpress/date to devDependencies for testing

* [not verified] remove mocking function, use apiFetch against new endpoint

* [not verified] temp commit

* [not verified] complete migration to wordpress/data

* [not verified] use first available response

* [not verified] return null while loading or without a response

* [not verified] fixup project versions

* fix pnpm lock file

---------

Co-authored-by: Kuba Birecki <kuba.birecki@crystalplanet.io>

matticbot committed:
Update renovatebot/github-action action to v34.154.2 (#29257)

* Update renovatebot/github-action action to v34.154.2
* Set `osvVulnerabilityAlerts` true
* Add `pnpmDedupe` to `postUpdateOptions`
* Include `group:definitelyTyped` preset

---------

Co-authored-by: Renovate Bot <bot@renovateapp.com>
Co-authored-by: Brad Jorsch <brad.jorsch@automattic.com>

retrofox committed:
VideoPress: add Caption button to the VideoPress video block toolbar (#29233)

* [not verified] minor caption field changes

* [not verified] add caption button to the block toolbar

* [not verified] move caption handling to main edit fn

* [not verified] focus caption element when adding

* [not verified] changelog

retrofox committed:
VideoPress: fix requesting video data on Simple sites (#29261)

* ensure global mode when requesting video data

* changelog

thingalon committed:
Fix Lazy Loading in Safari (#29243)

* Add a fix for Safari lazy loading

* changelog

---------

Co-authored-by: Mark George <thingalon@gmail.com>

cometgrrl committed:
adds a notification task to notify KitKat for high and blocker bugs (#28904)

Co-authored-by: Jeremy Herve <jeremy@jeremy.hu>

tommusrhodus committed:
Fixes search results extended view for searches hitting `isMultiSite` (#29179)

* Merge branch 'trunk' into update/add-date-to-search-results

* Fixes check against isMultiSite

Also removes erroenous checks against is_multisite()

* Allows multisite users to toggle post date on/off correctly

---------

Co-authored-by: sdixon194 <steve.dixon@automattic.com>

thingalon committed:
[Super Cache] Add missing changelog entry. (#29264)

* Add missing changelog entry

* changelog

---------

Co-authored-by: Mark George <thingalon@gmail.com>

thingalon committed:
[Boost] Use snake_case on tracks event props (#29111)

* Snake_case props on tracks events

* Snake_case props on tracks events

* changelog

---------

Co-authored-by: Mark George <thingalon@gmail.com>
Co-authored-by: Peter Petrov <peter.petrov89@gmail.com>

dereksmart committed:
SSO: Add message to logout notice that gives a heads up to also log out of wpcom if they're on a shared computer. (#29235)

* Add a heads up to the logout notice when using SSO to also log out of wpcom

* changelog

* period

* Add link to wp.com

Co-authored-by: Jeremy Herve <jeremy@jeremy.hu>

---------

Co-authored-by: Brandon Kraft <public@brandonkraft.com>
Co-authored-by: Jeremy Herve <jeremy@jeremy.hu>

donnchawp committed:
Jetpack Forms: Add/week month props on sent message tracks event (#28999)

* Add week and month props to contact_form_block_message_sent Tracks event

* changelog

* Remove deprecated is_block check, add it as a property

* Add no_post property if there's no parent post

* Update for changelog

* Merging again

* Use bools rather than integers for stat properties, props @jeherve

* Updating versions

dereksmart committed:
E2E: Temporarily disable Gutenberg subscribe block test (#29280)

* E2E: Temporarily disable gutenberg subscribe block test

* changelog

kozer committed:
Add LoadingPlaceholder component (#29270)

* Add: Add loading placeholder component

* changelog

* Update: update class name to avoid potential clash with other placeholder components

* Update: change jp-gray to hex color as jpitm css is not available when we use it inside blocks

* Update: remove previous changes as they are not necessary

* Update: bump version

jeherve committed:
Sharing: remove unused variable (#29275)

dereksmart committed:
E2E: Fix connection tests (#29284)

* E2E: Fix connection tests

* [not verified] update selector

* [not verified] update element selectors

ice9js committed:
Add default values for jetpack forms' CSS veriables (#29236)

* Add default values for jetpack forms' CSS veriables

* changelog

---------

Co-authored-by: sdixon194 <steve.dixon@automattic.com>

jhnstn committed:
[not verified] Merge branch 'trunk' into rnmobile/update-vp-player
